### ISSUE TITLE
NPM and forwarding client IP 2

### ISSUE THREAD

**Issue by tobias86aa** [webmin/webmin#2070]
Related to this question: https://github.com/webmin/webmin/issues/2068

How can i get Webmin to see the real visitor IP address of a client when it is behind a reverse proxy? Right now it only see this ip address: 192.168.1.118

**Comment by iliajie** [comment]
Hello,

Have a look at **Webmin ⇾ Webmin Configuration: IP Access Control** page and **Trust remote IP address provided by proxies?** option.

**Comment by tokar86a** [comment]
> Hello,
> 
> Have a look at **Webmin ⇾ Webmin Configuration: IP Access Control** page and **Trust remote IP address provided by proxies?** option.

Yes i have and it still giving me the same ip address of 192.168.1.118 when i turn that on.

**Comment by iliajie** [comment]
> Yes i have and it still giving me the same ip address of 192.168.1.118 when i turn that on.

You need to make sure that your NPM sets either `x-forwarded-for` or `x-real-ip` or `true-client-ip` headers.

**Comment by tokar86a** [comment]
> > Yes i have and it still giving me the same ip address of 192.168.1.118 when i turn that on.
> 
> You need to make sure that your NPM sets either `x-forwarded-for` or `x-real-ip` or `true-client-ip` headers.

From what i can see it set the following information

$_SERVER['HTTP_X_REAL_IP'] | 92.35.60.**
-- | --
$_SERVER['HTTP_X_FORWARDED_FOR'] | 92.35.60.76, 92.35.60.**

**Comment by iliajie** [comment]
> From what i can see it set the following information

I'm not sure what's wrong exactly on your side. Although, the code in Webmin that's responsible is here:

https://github.com/webmin/webmin/blob/master/miniserv.pl/#L1449-L1490

**Comment by tokar86a** [comment]
Can i somehow sett the trusted ip address? 

When i add following line to my apache configuration it seems to have solve the problem there.

RemoteIPHeader X-Forwarded-For
# replace IP with the remote server you trust
RemoteIPInternalProxy 192.168.1.118

**Comment by tokar86a** [comment]
@iliajie can this be something related? https://github.com/NginxProxyManager/nginx-proxy-manager/issues/3431#issuecomment-1887398994

**Comment by tokar86a** [comment]
@iliajie is that something that can be done here? Or is that something that happens whit you enable that option? https://github.com/webmin/webmin/issues/2070#issuecomment-1873880107

**Comment by iliajie** [comment]
> How can i get Webmin to see the real visitor IP address of a client when it is behind a reverse proxy?

I think I misread your original question. Take a look at **Reverse-resolve connected IP address?** option in **Webmin ⇾ Webmin Configuration: Ports and Addresses** page.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.
